### PR TITLE
[Merged by Bors] - Impl Reflect for PathBuf and OsString

### LINKED ISSUE #6166: Implement Reflect for PathBuf and OsString
## What problem does this solve or what need does it fill?

We currently have `Reflect` impl for `String`, but not for `PathBuf`. It would be great to have it implemented for `PathBuf` too.

### PR DESCRIPTION
**Pull request by Shatur**
# Objective

`Reflect` impl is missing for `PathBuf` and `OsString`. Closes #6166.

## Solution

Add implementations.

---

### Added

`Reflect` impls for `PathBuf` and `OsString`.

### CODE REVIEW

**Comment by Shatur** [comment]
@alice-i-cecile I added implementation for `OsString` only on platforms that can serialize it [as in serde](
https://github.com/serde-rs/serde/blob/3ffb86fc70efd3d329519e2dddfa306cc04f167c/serde/src/de/impls.rs#L1732).

**Comment by alice-i-cecile** [comment]
Great, thanks for the context there. Can you add that as a comment?

**Comment by Shatur** [comment]
Done!

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_reflect/src/impls/std.rs [hunk 1]
```diff
@@ -62,6 +66,11 @@ impl_reflect_value!(NonZeroU16(Debug, Hash, PartialEq, Serialize, Deserialize));
 impl_reflect_value!(NonZeroU8(Debug, Hash, PartialEq, Serialize, Deserialize));
 impl_reflect_value!(NonZeroI8(Debug, Hash, PartialEq, Serialize, Deserialize));
 
+// Only for platforms that can serialize it as in serde:
+// https://github.com/serde-rs/serde/blob/3ffb86fc70efd3d329519e2dddfa306cc04f167c/serde/src/de/impls.rs#L1732
+#[cfg(all(feature = "std", any(unix, windows)))]
```
Comment (tguichaoua) [inline crates/bevy_reflect/src/impls/std.rs:71]:
```suggestion
#[cfg(any(unix, windows))]
```

File: crates/bevy_reflect/src/impls/std.rs [hunk 2]
```diff
@@ -11,6 +11,8 @@ use crate::utility::{GenericTypeInfoCell, NonGenericTypeInfoCell};
 use bevy_reflect_derive::{impl_from_reflect_value, impl_reflect_value};
 use bevy_utils::{Duration, Instant};
 use bevy_utils::{HashMap, HashSet};
+#[cfg(all(feature = "std", any(unix, windows)))]
```
Comment (tguichaoua) [inline crates/bevy_reflect/src/impls/std.rs:14]:
```suggestion
#[cfg(any(unix, windows))]
```

**Review by tguichaoua — CHANGES_REQUESTED** [review]
I think `feature = "std"` is used in `serde` because it's a `no_std` crate. It's not necessary here.

**Comment by Shatur** [comment]
Of course, thanks, fixed.

**Review by tguichaoua — APPROVED** [review]

**Review by MrGVSV — APPROVED** [review]

**Pull request closed without merging**